File: backend/app/schemas/collections.py
"""Request/response shapes for collections and document upload. No
``tenant_id`` field anywhere -- same rule as app/schemas/conversations.py."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.services.chunking import MAX_CHUNK_SIZE, MIN_CHUNK_SIZE


class CreateCollectionRequest(BaseModel):
    name: str


class CollectionResponse(BaseModel):
    id: UUID
    name: str
    created_at: datetime


class DocumentResponse(BaseModel):
    id: UUID
    collection_id: UUID
    filename: str
    content_type: str
    status: str
    error: str | None
    created_at: datetime


class CollectionDetail(CollectionResponse):
    documents: list[DocumentResponse]


class UploadDocumentQuery(BaseModel):
    """Ingestion-time controls (STEP 25) -- chunk size/overlap are fixed
    for a document once it's indexed; changing them means re-uploading,
    never a query-time toggle. Sent as form fields alongside the file
    (multipart/form-data), not JSON -- see app/api/collections.py."""

    chunk_size: int = Field(default=1000, ge=MIN_CHUNK_SIZE, le=MAX_CHUNK_SIZE)
    overlap: int = Field(default=150, ge=0)

File: frontend/src/api/collections.ts
import { apiFetch, apiJson } from "./client";
import type { CollectionDetail, CollectionSummary, DocumentInfo } from "./types";

export async function createCollection(tenantId: string, name: string): Promise<CollectionSummary> {
  return apiJson<CollectionSummary>("/api/collections", tenantId, {
    method: "POST",
    body: JSON.stringify({ name }),
  });
}

export async function listCollections(tenantId: string): Promise<CollectionSummary[]> {
  return apiJson<CollectionSummary[]>("/api/collections", tenantId);
}

export async function getCollection(tenantId: string, collectionId: string): Promise<CollectionDetail> {
  return apiJson<CollectionDetail>(`/api/collections/${collectionId}`, tenantId);
}

/**
 * multipart/form-data upload -- the file plus ingestion-time controls
 * (chunk size/overlap, see app/schemas/collections.py::UploadDocumentQuery
 * for why these are form fields here and not query-time JSON like
 * top_k/similarity_threshold are on the chat request).
 */
export async function uploadDocument(
  tenantId: string,
  collectionId: string,
  file: File,
  chunkSize: number,
  overlap: number,
): Promise<DocumentInfo> {
  const form = new FormData();
  form.append("file", file);
  form.append("chunk_size", String(chunkSize));
  form.append("overlap", String(overlap));

  const response = await apiFetch(`/api/collections/${collectionId}/documents`, tenantId, {
    method: "POST",
    body: form,
  });
  return response.json() as Promise<DocumentInfo>;
}
